iceoryx_posh/cmake/IceoryxPoshDeployment.cmake

```diff
@@ -27,46 +39,102 @@ if(NOT IOX_COMMUNICATION_POLICY)
-    set(IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY 8)
+# Refer to iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/ipc_channel.hpp
+# for info why this is needed.
+if(APPLE)
+    set(IOX_MAX_RUNTIME_NAME_LENGTH_DEFAULT 98)
+else()
+    set(IOX_MAX_RUNTIME_NAME_LENGTH_DEFAULT 100)
 endif()
-message(STATUS "[i] IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY:" ${IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY})
 
-if(NOT IOX_MAX_PUBLISHER_HISTORY)
-    set(IOX_MAX_PUBLISHER_HISTORY 16)
-endif()
-message(STATUS "[i] IOX_MAX_PUBLISHER_HISTORY:" ${IOX_MAX_PUBLISHER_HISTORY})
-
-if(NOT IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY)
-    set(IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY 256)
-endif()
-message(STATUS "[i] IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY:" ${IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY})
+configure_option(
+    NAME IOX_MAX_PUBLISHERS
+    DEFAULT_VALUE 512
+)
+configure_option(
+    NAME IOX_MAX_SUBSCRIBERS
+    DEFAULT_VALUE 1024
+)
+configure_option(
+    NAME IOX_MAX_INTERFACE_NUMBER
+    DEFAULT_VALUE 4
+)
+configure_option(
+    NAME IOX_MAX_SUBSCRIBERS_PER_PUBLISHER
+    DEFAULT_VALUE 256
+)
+configure_option(
+    NAME IOX_MAX_CHUNKS_ALLOCATED_PER_PUBLISHER_SIMULTANEOUSLY
+    DEFAULT_VALUE 8
+)
+configure_option(
+    NAME IOX_MAX_PUBLISHER_HISTORY
+    DEFAULT_VALUE 16
+)
+configure_option(
+    NAME IOX_MAX_CHUNKS_HELD_PER_SUBSCRIBER_SIMULTANEOUSLY
+    DEFAULT_VALUE 256
+)
+configure_option(
+    NAME IOX_MAX_PROCESS_NUMBER
+    DEFAULT_VALUE 300
+)
+configure_option(
+    NAME IOX_MAX_NODE_NUMBER
+    DEFAULT_VALUE 1000
+)
+configure_option(
+    NAME IOX_MAX_NODE_PER_PROCESS
+    DEFAULT_VALUE 50
+)
+configure_option(
+    NAME IOX_MAX_SHM_SEGMENTS
+    DEFAULT_VALUE 100
+)
+configure_option(
+    NAME IOX_MAX_NUMBER_OF_MEMPOOLS
+    DEFAULT_VALUE 32
+)
+configure_option(
+    NAME IOX_MAX_NUMBER_OF_CONDITION_VARIABLES
+    DEFAULT_VALUE 1024
+)
+configure_option(
+    NAME IOX_MAX_NODE_NAME_LENGTH
+    DEFAULT_VALUE 100
+)
+configure_option(
+    NAME IOX_MAX_ID_STRING_LENGTH
+    DEFAULT_VALUE 100
+)
+configure_option(
+    NAME IOX_MAX_RUNTIME_NAME_LENGTH
+    DEFAULT_VALUE ${IOX_MAX_RUNTIME_NAME_LENGTH_DEFAULT}
+)
+configure_option(
+    NAME IOX_MAX_RESPONSES_PROCESSED_SIMULTANEOUSLY
+    DEFAULT_VALUE 16
+)
+configure_option(
+    NAME IOX_MAX_RESPONSE_QUEUE_CAPACITY
+    DEFAULT_VALUE 16
+)
+configure_option(
+    NAME IOX_MAX_REQUEST_QUEUE_CAPACITY
+    DEFAULT_VALUE 1024
+)
+configure_option(
+    NAME IOX_MAX_CLIENTS_PER_SERVER
+    DEFAULT_VALUE 256
+)
 
 # note: don't change IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS value because it could break the C-Binding
-#if(NOT IOX_MAX_NUMBER_OF_NOTIFIERS)
+#configure_option(
+#    NAME IOX_MAX_NUMBER_OF_NOTIFIERS
+#    DEFAULT_VALUE 256
+#)
 set(IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS 256)
-#endif()
-#message(STATUS "[i] IOX_MAX_NUMBER_OF_NOTIFIERS:" ${IOX_MAX_NUMBER_OF_NOTIFIERS})
+
```

Comment:
I think this is a remnant of the v2 release crunch time. AFAIK the note is not true any more and we could remove `INTERNAL`. @elfenpiff @dkroenke please correct me if I'm wrong